snippets/core/system-text-json/csharp/WeatherForecastRequiredPropertyConverter.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SystemTextJsonSamples
+{
+    public class WeatherForecastRequiredPropertyConverter : JsonConverter<WeatherForecast>
+    {
+        public override WeatherForecast Read(
+            ref Utf8JsonReader reader,
+            Type type,
+            JsonSerializerOptions options)
+        {
+            // Don't pass in options when recursively calling Deserialize.
+            WeatherForecast value = JsonSerializer.Deserialize<WeatherForecast>(ref reader);
+
+            // Check for required fields set by values in JSON
+            if (value.Date == default)
```

Comment:
I think this could be problematic. If the `Date` did exist in the JSON payload, but just happen to be the default value, then we still throw an exception.

For example, if `jsonString` was the following (or you serialize a POCO containing  `DateTime date = default`).
```json
{
  "Date": "0001-01-01T00:00:00+00:00",
  "TemperatureCelsius": 25,
  "Summary": "Hot"
}
```

Not sure if there is a great way to do this without using `nullable`, at least for value types like `DateTime`, int, etc.

Reply:
Do you think it better to add some text warning about the case where the date has the default value in the JSON, or  remove  the converter as a workaround from the doc?

Reply:
I can't think of a great/general purpose workaround for `Required` properties atm. **For now, text warning about the case is probably fine.** I will have to think about the scenario some more. One option could be to create a sentinel default value and check against that (like for an int property that is known to be positive, choose -1), but again, you wouldn't know if the payload actually contained that value or it just didn't exist.

Even for reference types (or nullable), how do you differentiate between the property not existing vs one that exists but is explicitly set to null.

Reply:
@steveharter, @layomia - do you have any ideas or suggestions?